crates/flox-rust-sdk/src/models/manifest.rs

```diff
@@ -86,6 +93,40 @@ pub fn insert_packages(
     })
 }
 
+/// Remove package names from the `[install]` table of a manifest
+pub fn remove_packages(
+    manifest_contents: &str,
+    pkgs: impl Iterator<Item = String>,
+) -> Result<Document, TomlEditError> {
+    debug!("attempting to remove packages from the manifest");
+    let mut toml = manifest_contents
+        .parse::<Document>()
+        .map_err(TomlEditError::ParseManifest)?;
+    match toml.entry("install") {
+        toml_edit::Entry::Occupied(ref mut existing_installs) => {
+            debug!("editing existing [install] table");
+            if let Item::Table(ref mut installs) = existing_installs.get_mut() {
+                for pkg in pkgs {
+                    debug!("checking for presence of package '{pkg}'");
+                    if !installs.contains_key(&pkg) {
+                        debug!("package '{pkg}' wasn't found");
+                        return Err(TomlEditError::PackageNotFound(pkg.clone()));
+                    } else {
+                        installs.remove(&pkg);
+                        debug!("package '{pkg}' was removed");
+                    }
+                }
+            } else {
+                return Err(TomlEditError::MalformedInstallTable(
+                    existing_installs.get().type_name().into(),
+                ));
+            }
+        },
+        toml_edit::Entry::Vacant(_) => return Err(TomlEditError::MissingInstallTable),
+    };
```

Comment:
suggestion: flatten similar to suggestion in install PR https://github.com/flox/flox/pull/339/files#r1376263061

Reply:
Addressed in the `install` PR.